extensions/amp-analytics/0.1/variables.js

```diff
@@ -220,7 +220,7 @@ export class VariableService {
         // Expand string values further.
         p = this.expandTemplate(raw,
             new ExpansionOptions(options.vars, options.iterations - 1,
-                options.noEncode));
+                true /* noEncode */));
```

Comment:
First of all, this PR is to fix a double encoding issue caused by the recursive nature of function `expandTemplate`. if `expandTemplate` is called with `noEncode=false`, we make sure the following recursive calls will turn encoding off.

that's to say, we do encode the var in URL, thus it will introduce new vulnerability .

If you're talking about a potential existing vulnerability in production, i took a look, the only place where `expandTemplate` is called with `noEncode=true` is [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/amp-analytics.js#L265). which is to expand an element selector (e.g. 'amp-img#img1). since the result of the expansion will be used to select elements in DOM (document.querySelector('amp-img#img1')) rather than be applied for URL replacement, so I think it should be safe.